tests/tracking/_model_registry/test_model_registry_fluent.py

```diff
@@ -26,7 +26,16 @@ def test_register_model_with_runs_uri():
         "_create_model_version",
         return_value=ModelVersion("Model 1", "1", creation_timestamp=123),
     )
-    with get_uri_patch, create_model_patch, create_version_patch:
+    is_directory_patch = mock.patch(
+        "mlflow.store.artifact.runs_artifact_repo.RunsArtifactRepository._is_directory",
+        return_value=True,
+    )
```

Comment:
what is this failing?

Reply:
https://github.com/mlflow/mlflow/actions/runs/13235482646/job/36939495724#step:13:536 invalid bucket name

Reply:
why patching `_is_directory` fixes the issue?

Reply:
Because _is_directory internally uses list_artifacts, and we're using S3 artifact repo. `s3:/path/to/source` doesn't include a valid bucket, but anyways it's not a real s3 path so even if it includes a bucket list_artifacts still fail, so I just mocked _is_directory to skip the check

Reply:
I think this test relies on mock too much. Let's remove them. We can log a model and register it with runs URI like this.

```python
def test_register_model_with_runs_uri(tmp_path):
    mlflow.set_tracking_uri(tmp_path.joinpath("mlruns").as_uri())

    class MyPyFuncModel(mlflow.pyfunc.PythonModel):
        def predict(self, context, model_input):
            return [1, 2, 3]

    with mlflow.start_run() as run:
        mlflow.pyfunc.log_model("model", python_model=MyPyFuncModel())

    register_model(f"runs:/{run.info.run_id}/model", "Model 1")
    version = MlflowClient().get_model_version(name="Model 1", version="1")
    assert version.name == "Model 1"
```

Reply:
Updated tests in https://github.com/mlflow/mlflow/pull/14572 against master, this PR should pass after that's merged